Zombie burner

Night-Pryanik opened the pull request:
Added zombie burner monster.
- It has a 10 percent chance to drip napalm from the damaged tank of its flamethrower;
- it has a 10 percent chance to explode in a fireball on its death;
- it will drop ordinary flamethrower on normal death. Half of the time flamethrowers' tanks will be filled, and if they do, they will have random charges on napalm in them;
- it will spawn in the same places as other military zombies do.

thethunderhawk commented:
Oh man, that’s gonna start some forest fires for sure. Sounds fun.
Those military outposts are getting ever more deadly, probably a good thing considering how easy they are to cheese for their loot.

Wishbringer commented:
This will probably work great with #23699, though its an evil idea. :)

Duxas commented:
Too deadly monster, for a fight in cramped space. At least, make him parroted some phrases, to prevent players sudenly struck onto him.

Theundyingcode commented:
I suggest adding him to the explosive zombie blacklisting mod if you haven't already done so.

John-Candlebury commented:
Maybe it should have the tag to avoid patching through fire?

Else it might attack the player and immediately suicide from walking in its own flames,

pisskop commented:
I disagree with this mon.

from a lore standpoint.
from a gameplay standpoint.
from a consistency standpoint.

Night-Pryanik commented:
@pisskop what's your points?

thethunderhawk commented:
From a lore standpoint, it’s a guy with a flamethrower (presumably they were using it to try to kill some zeds) who has been zombified and is now using that weapon against you. It’s not a crazy leap from a lore standpoint. 

From a gameplay standpoint it adds a flame-based enemy. Problems posed by fire provide interesting challenges from a gameplay perspective. Granted, it might be too deadly or not fun, but I don’t see any reason to assume that it can’t be balanced in a way that makes it fun.

From a consistency perspective I agree, it’s weird that it’s the only zed that intelligently uses a weapon, but maybe that could be explained away with a well crafted description? For example, maybe the RM451 comes with an arm brace that could stay attached on a dead guy, leaving the trigger permanently in its grip?

Night-Pryanik commented:
There's a zombie grenadier (in two variants) and a zombie bio-operator that could use their weapons.

thethunderhawk commented:
The grenadier simply opens pouches according to its description. The bio-operator is using a martial arts bionic that’s tied directly into it’s brain.

Mecares commented:
Some ideas: 
The name sounds kinda generic it is descriptive which is of course good but I would personally like something like Hellfire Zombie or Inferno Zombie or Scorching Zombie or Flameburst Zombie
Lootwise I would consider basing it on a Hazmat Zombie or a Firefighter Zombie while adding a very high chance to drop lighters. Also butchering it could give you a chance to obtain Thermal Dissipation Cbms or the Mini-Flamethrower Com. Also I would give it a chance to drop jerry cans and fire extinguishers.

thethunderhawk commented:
Oh man that’s the ticket: it has a thermal dissipation CBM, just let it path right through the flames with some fire resistance and mention the CBM in the description.

Night-Pryanik commented:
While I don't mind having some CBMs in it, I imagined it as an ordinary soldier who was given the flamethrower, maybe even complete rookie as most other more experienced soldiers were dead or zombiefied.

pisskop commented:
* Gameplay wise.  Fire, as it is now, is untenable.
- It burns leather and cotton and plastic as if they were paper.
- As a flamethrower special attack, it is ranged, and undodgeable.
- It causes incredible destruction to cities and forests and even car hulls take damage from flame.
- Fire, as a status effect, cannot easily be stopped.  It will burn for quite a while, causing pain and damage beyond what is reasonable.

* Lore-wise
- Fire is the anti-zed.
- Zeds are not smart enough to use weapons.  Grenadiers actually dont 'use' their weapons.  They are actually rigged, in life, to quickly disperse their load.  And in death they use them as well.  I agree that grenadiers also go against the grain, as well.  Lets make a robot that uses hacks and drones instead of a zed.
- Zeds shouldn't use attacks that will kill themselves, or unreasonable swathes of their allies.
- Zeds should not be capable of destroying a base and all the hard-fought loot inside just by not being noticed early enough.  Hulks don't even do this.


~Fire should be nerfed first.
~Zeds should do heat damage, not use a ranged attack.  Chance to dodge, for one.
~Zeds should not set you on fire.
~Zeds should not kill their allies or destroy bases with standard attacks.
~This zeds sounds niche anyway; maybe a mod with a new location would work?

Heat damage currently simulates flame.  This should, imo, be fixed before we add a flamey zombie.  Let it actually increase BP heat.

Leland commented:
Gotta say that @pisskop makes some salient points on this. I've played around with flame-based attacks on monsters before (see: tribots) and experienced similar notes.

It'd be worthwhile to do some playtesting of this PR with #23699 on top and see if things are still insane.

Shoes01 commented:
A _long_ time ago, I had begun adding code for "radiant temperature". For monsters, it gave them a hot aura flag and used their proximity to increase the bp warmth of the player. It seems somewhat related to the topic at hand, so [here](https://github.com/Shoes01/Cataclysm-DDA/blob/adce625e149010e49838f51b18c9ee2d632367ed/src/game.cpp#L1695) is the code, in case it comes in handy.

FulcrumA commented:
I'd agree that the zombie using a weapon is kinda iffy. Grenadiers are suggested to do it practically reflectivelly, just flinging stuff from the pouches, while bio-operators are part-machines and the weapon use may be machine-assisted.

I would mind i tless if said zombie would semi-randomly create puffs of fire here and there to suggest it has almost no control over its weapon. Though I need such specialized zombies less in general and would rather see it as optional/mod thing. Alternatively, I wouldn't mind it as something thrown as a very rare thing into labs and made a step closer to bio-operators.

thethunderhawk commented:
Those points are totally valid yeah. For one, maybe both this and the grenadier ought to be bots, that way we can move away from the tech based zed in general, leaving the bionic zeds as the only ones left able to use technology. But that’s another PR.
I am curious to see how #23699 changes things. IRL flamethrowers can use thin fuels to create a flame that doesn’t stick to things, the kind that if modeled correctly wouldn’t instantly be setting everything around it on fire.

Night-Pryanik commented:
> I suggest adding him to the explosive zombie blacklisting mod if you haven't already done so.

Done in https://github.com/CleverRaven/Cataclysm-DDA/pull/23729/commits/b4da040f1d458ee9cf3112ed1d2c72730f020fdd.

thethunderhawk commented:
@Night-Pryanik I think I’m lore terms you’re making good points, but we need to consider what is actually fun to play. Getting set on fire with no recourse or having that super valuable endgame skill book unavoidably go up in flames is not fun. We’ve gotta find a way to deal with that while maintaining the sense of realism you’re talking about.

Night-Pryanik commented:
I think we are just too accustomed to have very few ranged threats in the game and don't want to change the status quo.

> Getting set on fire with no recourse or having that super valuable endgame skill book unavoidably go up in flames is not fun.

We already have a lot of ways to deal with fire: waiting without moving, stepping in water, using fire extinguishers, wearing fire-protecting gear, getting fire-protecting mutations, just avoiding fire at last.

thethunderhawk commented:
>too accustomed to ranged threats and don’t want to change the status quo

I disagree. I want this monster to work, but it needs to be done in a way that doesn’t totally disrupt gameplay.

>waiting without moving, water, fire extinguisher, fire protective clothing, avoidance

Most of those still result in massive unavoidable pain, and none of those deal with all the buildings and loot in the area being destroyed.

Duxas commented:
> It can be dodged with uncanny dodge. I find it dubious that normal human can dodge flame spray like that.
Nope. It can't.

Night-Pryanik commented:
> Most of those still result in massive unavoidable pain

Do tell me, fire should or should not cause pain? Especially fire originated from flamethrower, which is designed to wreck chaos, mayhem and massive unavoidable pain.

> all the buildings and loot in the area being destroyed.

I don't understand, what's wrong with this? Fires _will_ happen, sooner or later, in abandoned cities. This is, let's say, their normal behavior.

thethunderhawk commented:
>fire should or should not cause pain?

Yeah, I’m saying I agree with you regarding all your concerns with the realism side of things. Yes a flamethrower is a deadly weapon that ought to be able to set you on fire. Bear in mind though that this is a game, the gameplay is a priority. If you can’t make a flame zombie without disrupting big portions of the gameplay (for example by randomly losing the rewards for defeating a difficult group of monsters or being forced to play in an unfun way) the monster shouldn’t be included in the first place.

Night-Pryanik commented:
Okay, I still don't understand what's wrong with dying from dangerous monster? Let's take shocker brute as an example. 

It has ranged attack? Yes, and the flamer too.
It require some measures to deal with like having gear with specific protection? Yes, and the flamer too.
It is able to destroy structures that can lead to destruction of loot? Yes, and the flamer too.

So, losing to shocker brute is unfun? If no, then why losing to flamer should be unfun?

thethunderhawk commented:
Yeah I agree with you to some extent on those points, and if you can get the level of destruction down to shocker brute levels than you’ve solved the issue I’m describing. My fear is that as flame mechanics stand now this mon will be destroying entire buildings and everything inside as an afterthought.

SaltatorMortis commented:
don't forget that the firemechanic is in flux PR #23699 
you need to gather enoth "tinder" to get the fire above a lv1 Fire...
i guess its then a thing of tweaking how many tinder it will spread

ergo final tweaking after #23699 is merged...

Night-Pryanik commented:
> level of destruction

I still don't get the answer on my question about what's wrong with fires caused by zombie flamer destroying structures. You want fire from flamer but don't want this fire destroy structures and loot? Some form of heat damage that pisskop talked about?

thethunderhawk commented:
Yeah exactly, or hopefully #23699 will tone down the effects of the spot fires the flamethrower might cause. Like I said you could have a perfectly realistic flamethrower that doesn’t instantly turn everything it touches into an inferno. I think some damage to structures and loot might be alright, especially if it’s somewhat avoidable.

Leland commented:
I've begun playtesting with #23699 merged locally on top of this PR.

Notes so far:

* reaaaaally deadly. I'd recommend making it much, much slower – and hopefully this guy is quite rare.
* Unless you're a late game character, are lucky enough to have flame-retardant gear, or are lucky enough to be in a situation where you can put out fire within a few turns, you're getting killed, and sometimes in a single hit. That's tougher than a hulk!
* The standard approach of standing in place to put out the fire doesn't work here - flame zombie will just use that time to close, and kill.
* Melee is really only an option in firefighter gear. Run or gun, and gun is still scary because the zombie's got a ranged attack too.
* Guy also sometimes paths into his own fires, which'll kill him. Probably recommend giving him flameproof gear, which makes sense IMO
* "Flamer" [has different connotations](https://www.urbandictionary.com/define.php?term=flamer)  than what might be expected 😉
* If you get hit once and still manage to escape, you'll have lost a lot of your worn gear, which feels crappy
* With #23699, the fires started won't completely consume buildings - so that fear should be alleviated. I haven't seen a single building go up in smoke yet, actually. Maybe too nerfed?

All in all, terrifying and quite deadly.

Perhaps instead of nerfing, go the opposite direction. Make this zombie incredibly rare, buff the hell out of his health? Perhaps make him a boss in some part of the game?

thethunderhawk commented:
Very cool. 

>doesn’t use the ability against hostile mobs or npcs 

Well, at least it won’t be starting random fires all over the place, though it still seems like a consistency issue... 

-maybe it could glow in the dark due to the pilot light on the flamethrower
-it could be very slow, again due to the bulk of the equipment
-it could make a constant hissing noise

Still, the flamethrower ability could really use a rework.

@leland did you by any chance check on how different types of loot fare in the fires?

Theundyingcode commented:
>  did you by any chance check on how different types of loot fare in the fires?

Would a flaming flamethrower do bonus fire damage?

Night-Pryanik commented:
Nerfed mon by decreasing its speed and adding a luminance to spot it at night.

thethunderhawk commented:
Coolio, so the biggest concerns at this point are the survivabiliry of loot and the ridiculous effects of getting ambushed by this thing. Sounds like if you come around a corner and this guy is in your face your playthrough is over, but then again this isn’t the only mon that can have that affect at least this one is gonna be nice and slow right?

Leland commented on data/json/monsters.json:
```diff
@@ -1864,6 +1864,44 @@
+    "id": "mon_zombie_flamer",
+    "type": "MONSTER",
+    "name": "zombie flamer",
+    "description": "When government realized that bullets weren't able to stop zombies once and for all, they dispatched flamethrower units to burn the walking dead.  Too bad that zombies were too many and flamers were too scarce, so now you can see some unlucky undead soldier still trying to burn everything it encounters.",
```
Small rewrite:

> Realizing that bullets weren't enough, the government dispatched flamethrower units to burn the walking dead. Unfortunately, there were too many zombies and not enough flamers. The undead soldier you see before you is still trying to burn everything it encounters.

Leland commented on data/json/monsters.json:
```diff
@@ -1864,6 +1864,44 @@
+  {
+    "id": "mon_zombie_flamer",
+    "type": "MONSTER",
+    "name": "zombie flamer",
```
I'd recommend changing the name - "flamer" in American English typically refers to a flamboyant gay man

Rivet-the-Zombie commented:
>-maybe it could glow in the dark due to the pilot light on the flamethrower
-it could be very slow, due to the bulk of the equipment
-it could make a constant hissing noise (though I guess that would cause some weirdness with surrounding mobs)

Very cool ideas, but it really depends on which flamethrower this zed is carrying; the RM451 has no pilot light (it has a piezoelectric igniter instead) and is fairly light and compact.

kevingranade commented:
I agree with the points about this conflicting with lore, zombies aren't intended to use weapons in general, and a flamethrower in particular doesn't mesh with the rationale of the weapon triggering automatically.

For a fire-based ranged ranged attacker, a little bot would make some sense.

For a fire based zombie, it could have developed a way to produce a flammable chemical internally and then spray and ignite it.  I have trouble imagining this being as long ranged as a flamethrower, but that's fine because this thing seems to need a serious nerf anyway.

DracoGriffin commented:
The idea of it sounds fun, but practicality is nowhere near useful. It sounds like something that will be instantly banned by players. Counterargument for hulks: early on, players learn these have to be attacked at night or with decent ranged weapons. Later they learn they can be defeated with suitable gear/melee ability.

This monster just sounds straight-up game-breaking: there is nothing to prevent fire damage (how practical is having to go back to base/vehicle to switch gear to fire-resistant ones, if even possible for player given density of hordes and such) -- so if this monster attacks a base, it's not like you could have built specific furniture to prevent your stored items from being damaged -- whereas with hulks, at worst you'd have to dig out the wreckage and re-build the terrain (walls/storage/etc). Essentially, not too much is lost (other than containers being smashed and losing liquids) and everything can be repaired/replaced. And any gear on you that's not suitably resistant to fire will be lost -- and there's not really an option to fireproof items/gear, or an accessible way to prevent losing equipment. If anything, this enemy seems to be a way to control power creep, which is fine, but should be used in locations for that effect -- not meant to be encountered in the wild where it can wreak havoc on player's game performance or arbitrarily ruin X hours of gameplay that can't be prevented against (dying being an exception, but players that spend X hours or seasons to build a base shouldn't be punished for "oh well you weren't paying attention"). 

This creature however, everything that is not fire-resistant/immune would be destroyed and the meta effects (basically what happens to the game itself) as fire is not optimized and can cause serious slowdowns and issues (which is a reason I ban fungal monsters/buildings and recommend players to do as well after they've played with them enough or until destroying a fungal bloom/tower ACTUALLY removes all traces of fungal enemies/terrain infestation by X overmap radius or suitable). And this monster basically sounds like a similar issue -- without knowing, a player could have an entire city or forest go up in flames, which compromises the game's ability to run effectively.

I can't say how any of this will apply with the new fire PR #23699 but if anything, the damage should just be more along the lines of heat being applied rather than spreading actual fire. Also there should be containers or storage terrain that can be crafted/constructed that are fire proof / resistant, if even at a fraction of the storage. So this would just skew the meta more for mobile bases ("death mobiles") than stationary bases, especially since extinguishing fires is horribly impractical as-is. And this issue would just be exacerbated based on monster evolution scaling factor / spawn rate scaling factor world settings, so as the game goes in (if the player moves to new areas to avoid the fire zombie), they'd become more and more common.

pisskop commented:
this mon should not be a zombie, and we all know it.

fun idea for endgame challenge, needs to be kept out of regular rotation.  A S.T.A.L.K.E.R. mod would welcome a mon like this (not even, but at least that is semi consistent with that lore), but it has a very reduced capacity to fit in in a non-destructive way.

thethunderhawk commented:
I gotta say I like Kevin’s suggestion that it could be turned into a chemical weapon sort of thing. It would sure mesh better with the setting.

>without the player knowing a whole city or forest could go up in flames

@DracoGriffin we determined that the flamethrower ability is only activated against the player not other mons or NPCs, so it’s not likely that random fires are going to be set without the player being aware of it. I do agree though that the flamethrower ability ought to be reworked.

DracoGriffin commented:
>@DracoGriffin we determined that the flamethrower ability is only activated against the player not other mons or NPCs, so it’s not likely that random fires are going to be set without the player being aware of it. I do agree though that the flamethrower ability ought to be reworked.
Maybe it’s a super late stage evo of something? Bloaters and spitters both already have some freaky chemical reactions going on inside them...

I don't mind that it's a zombie, I'm not concerned what the reasoning is (I leave that to others), so much as the issue of creating actual fire, rather than a simulated burning/fire effect.

That'd be like a monster in Dwarf Fortress that purposely starts catsplosions, even worse if it's done on an undiscovered z-level, so you end up losing your save to major slowdown. That's what I'm concerned with here.

Regardless of who the monster targets, even if it is specifically the player, the wanton damage and calculation the results from major wild fires is unoptimized and extremely difficult to stop or put out. It is much "safer" to just cause heat damage (that is reduced by fire resistance on gear), until a time fire is coded to be less of a strain on a system.

So for instance, if the player encounters this monster and manages to escape, how much of that area is going to be destroyed or become useless to even come back to? So then if you go to another area, and can't kill the fire zombie(s) and that area becomes burned to the ground, and the cycle continues until you start a new world, and same issues again, until finally you just blacklist the fire zombie and play like normal. In contrast to grenadiers and shocker zombies/brutes, their presence may cause a player to turn around and avoid the area but it's temporary, since they can choose to go back and kill the zombies if/when they are stronger. This isn't really an option with the fire zombie: there generally is very little reason to return to an area that has been burned to the ground. This is why this should be an enemy reserved to more controlled areas for fire damage (like labs), and basically if you're not ready to kill it, guess what, you lose out on whatever sweet loot it was protecting or was nearby -- not "you encounter this in the alley behind a liquor store and now this fire is spreading everywhere, oh well, time to find a new city".

So roadblocks that cause me to instantly blacklist this monster in my games (and by extension, telling others to do the same):

- Few options to protect yourself or your items (or your base/stored items) from fire damage.
- Fire is poorly optimized and will slow down saves when loaded in reality bubble. More of these monsters, the bigger the potential issue.
- Little payoff for killing one, instead of just avoiding it. You can get inactive type hacks to disassemble (for parts or learning to craft) from grenadiers and good CBMs from shockers with decent survival skill.

thethunderhawk commented:
@DracoGriffin according to post #23699 play tests by @leland the fires are no longer spreading the way you’re describing.

pisskop commented:
> ittle payoff for killing one, instead of just avoiding it.

My concern here is that this is an excuse to add reliably spawning flame throwers instead of just . . . adding them to military bases or something.  It is true that as the military sent in special forces, flamers would be amoung them.  FEMA camps, anbc zoms, etc.  Heeey, an anbc zed sounds neato.

Theundyingcode commented:
In the cataclysm+ mod, there is a fire ball CBM IIRC.  Maybe use that?  Or, there are scorched/burnt zombies (idr the name) make the flamer their evolution.  you loose the excuse to add flamethrower drops but, it'll be easier to make it mesh well.

> ANBC zed does sound neato...
and bomb disposal zombies

Straume21 commented:
not really lore friendly... what about the flame zombie as a normal melee zombie, but drops a flamethrower as you said and has a chance to explode into flames like an incendiary grenade on death and when hit with enough damage because of the fuel tank it is wearing?

FulcrumA commented:
As I mentioned when talking with @Straume21 - I like this idea more, at least from the standpoint of lore. Various regular zombies but wearing potentially dangerous stuff seem interesting. I am ambivalent toward making it an incendiary grenade as it may lead to collateral which will simply kill fun rather than provide much of any particular challenge.

Night-Pryanik commented:
> For a fire-based ranged ranged attacker, a little bot would make some sense.

We already have a fire-throwing bot (tribot), and I'm not sure if making another, less powerful, bot with fire throwing is a worthwhile idea.

> For a fire based zombie, it could have developed a way to produce a flammable chemical internally and then spray and ignite it.

I don't know, maybe some sort of zombie alcoholic? Blob managed to change the metabolism of its host and now it stores alcohol in internal organs. Cause in life this alcoholic was a smoker too, it reflexively tries to light the cigarette (not even realizing there isn't one), and this way ignites the alcohol vapors. Hmm?

Either way, I don't want to convert this mon to do heat damage (I mean _only_ heat damage; heat melee damage in addition to flame throwing is ok).

kevingranade commented:
> Either way, I don't want to convert this mon to do heat damage (I mean only heat damage; heat melee damage in addition to flame throwing is ok).

I agree heat damage doesn't sound very fun.

My suggestion was, if you want to keep it as a flame-type zombie, describe it as creating/oozing flammable liquid, and have it somehow ignite it.  The end result is it would fire short jets of fire (just 2-3 squares probably) and probably have a flaming burst when killed.

Or, a little bit out of left field.  What if the flammable zombie is just a flammable boomer, and it relies on something like getting zapped by a shocker or set on fire to turn it into flame?
This giy would chase you around dribbling gasoline all over, and if it encounters fire everything goes up.  Alone they wouldn't be a big deal, but in tandem with a shocker they're beyond deadly.

Night-Pryanik commented:
Gasoline boomer sounds fun. We only have to think up how it was able to accumulate gasoline inside.

pisskop commented:
Hmmm?  Gas as it it now is entirely weak.  I tried to make a gas dispersing bomb for player use, and 40k of gas units will burn off with the same fervor as 1.

Im thinking that, if gas is properly flammable with the nerf, just make it a kind of chipmonk or pelican zed.  with a pouch for spitting ingested gas from.


so many cars are bereft of gas, and get destroyed, a zed that drinks gas is almost recycling.  if it can actually steal your gas, so much the better.

the endgame, lorewise, is to absorb more to make blobs.  Adding a code to make shockers zap these if they are both within range of you and the gaszed is doable too.

thethunderhawk commented:
While we’re spitballing, a “smoldering zombie” (evo of a smoker?) that does proximity heat damage could be interesting. Like a mobile lava patch guy. Could sync with the gas boomer being described. Or, get this: “fermenting zombie”. Highly flammable.

IMO Kevin’s first suggestion for a weaker flame attack on a more lore friendly zed is a good one.

SaltatorMortis commented:
this Lore friendy suggestion could work:
- Flamer spits pyrophoric liquid (air ignititing Liquid)(with Range defined Spashzones)
   - ignites only if the Liquid aggregates above a specified Volume and then only after a delay of a few Turns
   - soaks Clothes+Player if not protected with a Watertight Layer

      - ie Player needs two or more Hits to get soaked enough to trigger the ignition Countdown
which is interruptable if you drop some clothes/white yoursef dry+throw the towel away



> con
> - if you dont have a protecting Watertightlayer you need to micro the Escape (throw away/drop soaked things to prevent Ignition)

> pro
> - no fires if you don't idle for too long at one spot

Night-Pryanik commented:
Okay, reworked this mon. 
- Removed a `FLAMETHROWER` special attack;
- It has a 10 percent chance to drip napalm on its move;
- Removed chance to drop RM451 flamethrower on normal death. It will drop only ordinary flamethrower with pilot light that will light this mon in the dark;
- Flamethrower tanks will have a random number of charges in them.

I believe this way we was able to solve most issues with this mon. 

The more time this mon is walking on the battlefield, the more dangerous the battlefield becomes as it is being filled with napalm on the ground. Killing the _zombie burner_ can make a wildfire from all this napalm.

Testing shows that electricity attacks from the shockers do not set napalm (or gasoline) on fire, but I don't know if this happens IRL, but even if it is, I think it could be resolved in another PR.

Theundyingcode commented:
> It has a 10 percent chance to drip napalm on its move;

"I love the smell of burner Zs in the morning"  Please figure out how to add that somewhere.

thethunderhawk commented:
I’m sorry I just don’t think this is very consistent with the realism of the setting. Its gonna be dripping infinite napalm, the pilot light somehow doesn’t set off the vapor, and getting a kill with a paper wrapper can still set off a fiery explosion.

Straume21 commented:
The idea, correct me if I'm wrong, was that the zombie would have a chance
to explode into flames upon death, not a guaranteed explosion, and they
don't constantly drop napalm.. those are just ideas, not things that will
definitely be implemented. I think the idea of a zombie flamethrower (a
human that was carrying a flamethrower and fuel tank before they perished)
is reasonable. It's just a variant of a zombie soldier basically.

On Mon, May 28, 2018, 12:41 PM thethunderhawk <notifications@github.com>
wrote:

> I’m sorry I just don’t think this is very consistent with the realism of
> the setting. Its gonna be dripping infinite napalm, the pilot light somehow
> doesn’t set off the vapor, and getting a kill with a paper wrapper can
> still set off a fiery explosion.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/23729#issuecomment-392568051>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlOafyhRd5Eeojg7FB3-eo3TVt-MxKtbks5t3Ci_gaJpZM4T-M6J>
> .
>

Night-Pryanik commented:
I'd call constant napalm dripping and possible fire explosion upon death an acceptable gaming convention.

> The fact that it’s only a 10% chance of presenting a different challenge than a regular zed is a little odd too.

Why is that? Its tank is damaged. It shouldn't explode on every death, only on player's bad luck. It's not a regular zed in any way.

thethunderhawk commented:
>these are just ideas

Yes, but it’s also the current state of the PR. I definitely don’t have a problem with a zombie carrying a flamethrower if it’s implemented well.

>I’d call that an acceptable gaming convention

Yeah it’s definitely just my opinion, but not from a gameplay standpoint but from a lore and consistency standpoint I think this will be a square peg in a round hole.

> it’s not a regular zed in any way

Ok I’ll rephrase my concern: this zed is weird because of how unlucky you have to be for it to trigger its signature ability. At a 10% activation rate a player could fight several of these and never have the chance to learn that they can be super dangerous.

pisskop commented:
A 10% to end a players run is terrible.

Straume21 commented:
so just don't.. go near them? only fight them with ranged or when
sufficiently armored... a player with a lot of fire protection wouldn't get
gibbed by them. a zombie hulk has a 100% chance to end a young survivor's
run, yet we have those in...

On Mon, May 28, 2018, 4:31 PM pisskop <notifications@github.com> wrote:

> A 10% to end a players run is terrible.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/23729#issuecomment-392598967>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlOaf3D6vWPfLuNW5BO-k70AWiZHr9F_ks5t3F6FgaJpZM4T-M6J>
> .
>

Night-Pryanik commented:
I see nothing wrong with some gameplay mechanics possible being obscure to players until they discover how to deal with it themselves.
One good example for this: there seems to be nothing particularly bad with inhaling fungal spores - until later time. Player has to figure out that a) he should wear mouth gear with good environment protection to prevent inhaling; b) if he inhaled spores nevertheless, he has to take antifungal drugs to prevent loss of arms.
Hell, some players can die until they figure that they shouldn't light a fire in open field.

> A 10% to end a players run is terrible.

And no, fire explosion doesn't necessarily means player's death.

BorkBorkGoesTheCode commented:
Flamethrowers can explode into a fireball, but they are more likely to just explode with concussive force when their propellant tank ruptures.

pisskop commented:
My suggestion is that somebody actually play a test game where this critter is actively spawning from game start.  It spawning before a player has a full suit of flameproof armor or a means to flee is going to cause grief.  This griefing of the player is not going to go over well -look at grenadiers.  Similarly powerful monsters are regulated to special locations and situations, not the general zombie population pool.

thethunderhawk commented:
Oh jeez this isn’t already a Mil Outpost only spawn? I hadn’t noticed. This zed has new, powerful mechanics and is likely to cause major environmental damage and has potential to disrupt general gameplay, IMO it really ought to enter the game as a mil outpost spawn until there’s some serious player feedback. You wouldn’t just invent the grenadier and throw it in the middle of every town, this certainly has the potentional to be similarly disruptive and deadly.

thethunderhawk commented:
@borkborkgoesthecode that’s interesting where’d you learn that?

BorkBorkGoesTheCode commented:
>Flamethrowers can explode into a fireball, but they are more likely to just explode with concussive force when their propellant tank ruptures.

That I would mostly retract.

Night-Pryanik commented:
Ok, removed zombie burner spawn from all monster groups except for `GROUP_MAYBE_MIL`. Also added chance for it to spawn at military outpost. This should prevent player from encountering it at hospital or grocery next door.

thethunderhawk commented:
Various rework ideas that might gain more traction, in no particular order:
• all it does is explode on death maybe 50% of the time. Lorewise it’s tank ruptures and the pilot light ignites the fuel vapor.
• it’s a special zombie soldier that drops an active Molotov or incendiary grenade, lab/outpost/necropolis only.
• it’s a bombardier beetle style anthill boss
• it’s some kind of broken bot, with the currently listed mechanics. This would take some creativity lorewise.
• a late stage chemical based evo chain that splits the fuel leakage and the fiery death into separate, synergistic zeds. 
• it’s a shoggoth-style extradimentional mon that appears only in prisoner containment.
• the original idea, but the flamethrower mechanics are reworked.

I’m an advocate of anything that adds real content. I’d love a fire-based challenge that uses mons.
